superset/databases/commands/create.py

```diff
@@ -54,10 +50,9 @@ def run(self) -> Model:
             try:
                 TestConnectionDatabaseCommand(self._actor, self._properties).run()
             except Exception:
-                db.session.rollback()
-                stats_logger.incr(
-                    f"db_connection_failed.{database.db_engine_spec.__name__}"
-                )
+                with event_logger.log_context(action=f"db_connection_failed", engine=database.db_engine_spec.__name__):
```

Comment:
@mistercrunch we can pass the engine here as a separate kwarg, and downstream integrations can pick it up if they want to separately, rather than us passing it with the action. wdyt about that approach?
Also, just as an FYI, the strings that we're passing in these two files aren't exactly 1:1. Here we have to use the engine spec name, and in the test_connection file we pass the engine+driver combo. But the information is essentially the same.. any downstream integrations can still see the engine from the string that is passed in.